x-pack/filebeat/module/aws/cloudtrail/ingest/pipeline.yml

```diff
@@ -0,0 +1,130 @@
+processors:
+  - rename:
+      field: "message"
+      target_field: "event.original"
+      ignore_failure: true
+  - json:
+      field: "event.original"
+      target_field: "json"
+  - date:
+      field: "json.eventTime"
+      target_field: "@timestamp"
+      ignore_failure: true
+      formats:
+        - ISO8601
+  - rename:
+      field: "json.eventVersion"
+      target_field: "aws.cloudtrail.event_version"
+      ignore_missing: true
+  - rename:
+      field: "json.userIdentity.type"
+      target_field: "aws.cloudtrail.user_identity.type"
+      ignore_missing: true
+  - rename:
+      field: "json.userIdentity.userName"
+      target_field: "user.name"
+      ignore_failure: true
+  - rename:
+      field: "json.userIdentity.principalId"
+      target_field: "user.id" 
+      ignore_failure: true
+  - rename:
+      field: "json.userIdentity.arn"
+      target_field: "aws.cloudtrail.user_identity.arn"
+      ignore_failure: true
+  - rename:
+      field: "json.userIdentity.accountId"
+      target_field: "cloud.account.id"
+      ignore_failure: true
+  - rename:
+      field: "json.userIdentity.accessKeyId"
+      target_field: "aws.cloudtrail.user_identity.access_key_id"
+      ignore_failure: true
+  - rename:
+      field: "json.userIdentity.sessionContext.attributes.mfaAuthenticated"
+      target_field: "aws.cloudtrail.user_identity.session_context.mfa_authenticated"
+      ignore_failure: true
+  - date:
+      field: "json.userIdentity.sessionContext.attributes.creationDate"
+      target_field: "aws.cloudtrail.user_identity.session_context.creation_date"
+      ignore_failure: true
+      formats:
+        - ISO8601
+  - rename:
+      field: "json.userIdentity.invokedBy"
+      target_field: "aws.cloudtrail.user_identity.invoked_by"
+      ignore_failure: true
+  - rename:
+      field: "json.eventSource"
+      target_field: "event.provider"
+      ignore_failure: true
+  - rename:
+      field: "json.eventName"
+      target_field: "event.action"
+      ignore_failure: true
+  - rename:
+      field: "json.awsRegion"
+      target_field: "cloud.region"
+      ignore_failure: true
+  - rename:
+      field: "json.sourceIPAddress"
+      target_field: "source.address"
+      ignore_failure: true
+  - rename:
+      field: "json.userAgent"
+      target_field: "user_agent.original"
+      ignore_failure: true
+  - rename:
+      field: "json.errorCode"
+      target_field: "aws.cloudtrail.error_code"
+      ignore_failure: true
+  - rename:
+      field: "json.errorMessage"
+      target_field: "aws.cloudtrail.error_message"
+      ignore_failure: true
+  - rename:
+      field: "json.requestId"
+      target_field: "aws.cloudtrail.request_id"
+      ignore_failure: true
+  - rename:
+      field: "json.eventID"
+      target_field: event.id
+      ignore_failure: true
+  - rename:
+      field: "json.eventType"
+      target_field: event.category
+      ignore_failure: true
```

Comment:
`event.category` should never be populated with a source value.

In ECS 1.4, we're publishing the first allowed values for this field and other ([see this new section](https://www.elastic.co/guide/en/ecs/current/ecs-category-field-values-reference.html)).

If none of the values published so far for this field (more to come), this `event.category` field must be left empty.

So you should capture `json.eventType` in a custom field.

Reply:
> Thanks for making that adjustment. LGTM from ECS' POV :-)
> 
> You should likely get approval from another Beats maintainer as well, IMO. I only reviewed the mapping of the events.

Thanks for reviewing, that was a big help and gives me a little more confidence that I'm using ECS correctly.

It will be a bit before it is ready to merge, I still have to work out the S3 bucket parts.